tasmota/tasmota_xsns_sensor/xsns_116_ms5837.ino

```diff
@@ -0,0 +1,138 @@
+/*
+  xsns_14_sht3x.ino - SHT3X, SHT4X and SHTCX temperature and humidity sensor support for Tasmota
+
+  Copyright (C) 2022  Theo Arends, Stefan Tibus
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifdef USE_I2C
+#ifdef USE_MS5837
+
+#define MS5837_ADDR         0x76
+
+#define XSNS_116            116
+#define XI2C_91             91         // See I2CDEVICES.md
+/*********************************************************************************************\
+ * BlueRobotics Pressure Sensor
+ *
+ * This driver supports the following sensors:
+ * - BlueRobotics MS5837
+\*********************************************************************************************/
+
+#include <Wire.h>
+#include <MS5837.h>
+
+MS5837 sensor_ms5837;
+
+uint8_t ms5837Start = 0;
+float pressureOffset = 2.85;
+
+/********************************************************************************************/
+
+void MS5837init(void) {
+
+  if (I2cSetDevice(0x76)) {
+    TwoWire& myWire = I2cGetWire();
+
+    if(sensor_ms5837.init(myWire)) {
+      sensor_ms5837.setModel(sensor_ms5837.MS5837_02BA);
+      sensor_ms5837.setFluidDensity(997); // kg/m^3 (freshwater, 1029 for seawater)
+      ms5837Start = 1;
+      I2cSetActiveFound(MS5837_ADDR, "MS5837");
+    }
+  }
+}
+
+#ifdef USE_WEBSERVER
+const char HTTP_SNS_MS5837_DEFAULT[] PROGMEM =
+  "{s}MS5837 Temperature {m}%s " D_UNIT_DEGREE            "%c{e}{s}MS5837 Pressure {m}%s %s{e}";  // {s} = <tr><th>, {m} = </th><td>, {e} = </td></tr>
+const char HTTP_SNS_MS5837_INCHES_WATER[] PROGMEM =
+  "{s}Inches Water {m}%s in{e}";  // {s} = <tr><th>, {m} = </th><td>, {e} = </td></tr>
+char HTTP_SNS_MS5837_DUAL[118];
```

Comment:
Do not re-invent temperature and pressure HTTP_SNS. Use current ones.

And do not use inches of water. Use either hPA or mmHg

Reply:
Inches of water is an essential datapoint for the functionality being added. The dual usage of a BMP280 and MS5837 allows an atmospheric reading and underwater reading for pressure, which is used to calculate the depth of water of a reservoir. Suggestions for how to implement this would appreciated. I will rework the temperature and pressure HTTP_SNS to use the existing ones.

Reply:
Don't get it why inches should be needed.  1 inch = 2,54 cm = 0,0254 m 
So just a factor to change in a formula.

Reply:
Thank you for that clarification. That makes more sense, will use a conversion to adhere to metric standards for Tasmota.